### PR TITLE
Port fluentd to TF

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: images/fluentd/tests/main.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_exec_test" "test" {
+  digest = var.digest
+  script = "${path.module}/01-test.sh"
```
Comment (imjasonh) [inline images/fluentd/tests/main.tf:13]: Let's use `working_dir` so the test doesn't have to know about the structure of the repo above this directory.

Comment (mattmoor) [reply images/fluentd/tests/main.tf:13]: The files it is accessing are above this directory (`examples` is generally a sibling).  I think my preference would be to move `examples` under the test directory, but this confuses `monopod` right now.  I was hoping we could do some broader cleanup once we're off of `monopod matrix` 🤞 

I can change it, it just felt a little gross to specify a `working_dir` and then use `../examples` 🤢

Comment (mattmoor) [reply images/fluentd/tests/main.tf:13]: tl;dr I'd love to work towards relatively self-contained modules 🤞

Comment (mattmoor) [reply images/fluentd/tests/main.tf:13]: https://github.com/chainguard-images/images/issues/925

**Review by imjasonh — APPROVED** [review]

**Pull request merged**